### PR TITLE
Use torch.export.Dim in export tutorial instead of torch.export.dynamic_dim

### PR DESCRIPTION
**Pull request by williamwen42**
As title. Will not work for 2.1 release since `torch.export.Dim` is not in 2.1.

So I don't think PR can't be merged until the next PR, but I'm getting it out early.

### CODE REVIEW

File: intermediate_source/torch_export_tutorial.py [hunk 1]
```diff
@@ -255,91 +260,107 @@ def forward(self, x, y):
...
+# We can relax this constraint using the ``dynamic_shapes`` argument of
+# ``torch.export.export()``, which allows us to specify (using ``torch.export.Dim``)
+# which dimensions of the input tensors are dynamic.
 #
 # ..
-#     [TODO] link to doc of dynamic_dim when it is available
+#     [TODO] link to doc of Dim when it is available
+#
+# For each tensor argument of the input callable, we can specify a mapping from the dimension
+# to a ``torch.export.Dim``.
```
Comment (angelayi) [inline intermediate_source/torch_export_tutorial.py:271]: https://pytorch.org/docs/main/export.html#torch.export.Dim
